src/parsers.php

```diff
@@ -8,7 +8,7 @@
 
 function variable(): Parser
 {
-    return rtoken('/[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*/');
+    return rtoken('/\$[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*/');
```

Comment:
I'm sure you already know it, this is just an obvious reminder: variable has some other branches like https://3v4l.org/10bFt and the not so popular var var `$$$x` syntax.

Reply:
I seem to be missing something about `rtoken`. I would think the following would work:
```php
    return rtoken('/\$[\$]+[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*/');
```
but it doesn't. `$$a` is matched as `$a`.

Reply:
A single `\$+` works nicely according to https://regex101.com/r/WhuVPY/1

Reply:
Yeah, it should work, but `phpunit` disagrees:

```
There was 1 failure:

1) Yay\SpecsTest::testSpecs with data set "/Users/mduarte/Developer/code/MarcelloDuarte/yay/tests/phpt/expr/variable.phpt" (Yay\Test Object (...))
Failed asserting that format description matches text.
--- Expected
+++ Actual
@@ @@
 <?php

-expression {
-    $$a
+$expression {
+    $a
 }

 ?>
```

Reply:
Weird, this works here:

```php
use \Yay\{TokenStream, function rtoken};

$ts = TokenStream::FromSourceWithoutOpenTag('$a $bb $ccc');

token(T_OPEN_TAG))->parse($ts); // jump over <?php

var_dump(rtoken('/\$+[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*/')->parse($ts));
# T_VARIABLE($a)

var_dump(rtoken('/\$+[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*/')->parse($ts));
# T_VARIABLE($bb)

var_dump(rtoken('/\$+[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*/')->parse($ts));
# T_VARIABLE($ccc)

var_dump(rtoken('/\$+[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*/')->parse($ts));
# null
```

Just double checking, did you really use `/\$+[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*/` as the regexp? If so perhaps PCRE engine is behaving differently on our builds :thinking:

Reply:
Yeah, I did use it. It works fine with normal variables, but __not__ with variable variable:
```php
var_dump(\Yay\rtoken('/\$+[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*/')->parse(\Yay\TokenStream::fromSourceWithoutOpenTag('$$a')));
# null
```

Reply:
Ok, so there was one issue fixed with a2329f893.

But the real culprit is that PHP tokenizes variable variables as `[token('$'), token('$'), token('$a')]` instead of `[token('$$a')]`, hence why `rtoken` would never work.

The following works:

```php
use \Yay\{TokenStream};

use function \Yay\{chain, optional, repeat, token};

$variable = chain(optional(repeat(token('$')))->as('varvar'), token(T_VARIABLE)->as('var'));

$ts = TokenStream::FromSourceWithoutOpenTag('$a $$b $$$c');

var_dump($variable->parse($ts)); // Ast for $a
var_dump($variable->parse($ts)); // Ast for $$b
var_dump($variable->parse($ts)); // Ast for $$$c
var_dump($variable->parse($ts)); // null
```
:+1: